instaloader/nodeiterator.py

```diff
@@ -131,11 +132,13 @@ def __next__(self) -> T:
                 raise
             item = self._node_wrapper(node)
             if self._is_first is not None:
-                if self._is_first(item):
+                if self._is_first(item, self._first_item):
```

Comment:
I would prefer to pass `self._node_wrapper(self._first_node)` here as the second parameter, rather than storing `_first_item` besides `_first_node`. The node wrapper functions are all pure and quick, hence I think we should prefer the solution that requires less state managing.

Reply:
Fair point. And it makes the code simpler, since there's already a function for converting the `_first_node` with `_node_wrapper`, with proper `None`-checking.